src/idpi/operators/support_gis.py

```diff
@@ -0,0 +1,229 @@
+
+    # ... cache trigonometric operations
+    sin_lat = np.sin(deg2rad * grid.rlat)
+    cos_lat = np.cos(deg2rad * grid.rlat)
+    sin_lon = np.sin(norm_lon)
+    cos_lon = np.cos(norm_lon)
+    # ... compute latitude
+    arg1 = cos_np_lat * cos_lat * cos_lon + sin_np_lat * sin_lat
+    lat = rad2deg * np.arcsin(arg1)
+    # ... compute longitude
+    arg1 = (
+        sin_np_lon * (-sin_np_lat * cos_lon * cos_lat + cos_np_lat * sin_lat)
+        - cos_np_lon * sin_lon * cos_lat
+    )
+    arg2 = (
+        cos_np_lon * (-sin_np_lat * cos_lon * cos_lat + cos_np_lat * sin_lat)
+        + sin_np_lon * sin_lon * cos_lat
+    )
+    arg2 = xr.where(np.abs(arg2) < 1e-20, 1e-20, arg2)  # BUG: changes sign?
+    lon = rad2deg * np.arctan2(arg1, arg2) % 360
+
+    return lon, lat
+
+
+def geolatlon2swiss(
+    lon: xr.DataArray, lat: xr.DataArray
+) -> tuple[xr.DataArray, xr.DataArray]:
+    """Convert from geolatlon to swiss coordinates.
+
+    Parameters
+    ----------
+    lon : xarray.DataArray
+        longitude coordinates in the geolatlon CRS.
+    lat : xarray.DataArray
+        latitude coordinates in the geolatlon CRS.
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        x and y coordinates in the Swiss LV03 coordinate system.
+
+    Notes
+    -----
+    Approximate formula published by swisstopo, precision in the order of 1 meter
+
+    """
+    norm_lat = ((lat * 3.6) - 169.02866) / 10
+    lon = lon.where(lon < 180, lon - 360)
+    norm_lon = ((lon * 3.6) - 26.7825) / 10
+    y = (
+        200147.07
+        + 3745.25 * norm_lon * norm_lon
+        + norm_lat
+        * (
+            308807.95
+            + 76.63 * norm_lat
+            - 194.56 * norm_lon * norm_lon
+            + 119.79 * norm_lat * norm_lat
+        )
+    )
+    x = 600072.37 + norm_lon * (
+        211455.93
+        - 10938.51 * norm_lat
+        - 0.36 * norm_lat * norm_lat
+        - 44.54 * norm_lon * norm_lon
+    )
+    return x, y
+
+
+def vref_rot2geolatlon(
+    u: xr.DataArray, v: xr.DataArray
+) -> tuple[xr.DataArray, xr.DataArray]:
+    """Apply coordinate rotation to vector field.
+
+    When converting from rotated lat lon to geo lat lon, the
+    orientation of the grid changes and vector fields for which
+    the components are expressed in the grid unit vectors need
+    to be realigned.
+
+    Note that this function does not perform any regridding.
+
+    Parameters
+    ----------
+    u : xarray.DataArray
+        x component of the vector field w.r.t. a rotated lat lon grid.
+    v : xarray.DataArray
+        y component of the vector field w.r.t. a rotated lat lon grid.
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        x and y components of the vector field w.r.t. the geo lat lon coords.
+
+    """
+    if u.origin["x"] != 0.0 or v.origin["y"] != 0.0:
+        raise ValueError("The velocity fields must be destaggered.")
```

Comment:
I guess other fields other than velocities can be vectors? I'd maybe change valueerror to 
```suggestion
        raise ValueError("The vector fields must be destaggered.")
```

Reply:
Good point